cmd/syncthing/monitor.go

```diff
@@ -317,6 +327,94 @@ func restartMonitorWindows(args []string) error {
 	return cmd.Start()
 }
 
+// rotatedFile keeps a set of rotating logs. There will be the base file plus up
+// to maxFiles rotated ones, each ~ maxSize bytes large.
+type rotatedFile struct {
+	name            string
+	open            openFn
+	maxSize         int64 // bytes
+	maxFiles        int
+	minStatInterval time.Duration // wait at least this long between stat() calls
+	current         io.WriteCloser
+	lastStat        time.Time
+}
+
+type openFn func(name string) (io.WriteCloser, error)
+
+func newRotatedFile(name string, open openFn, maxSize int64, maxFiles int, minStatInterval time.Duration) *rotatedFile {
+	return &rotatedFile{
+		name:            name,
+		open:            open,
+		maxSize:         maxSize,
+		maxFiles:        maxFiles,
+		minStatInterval: minStatInterval,
+	}
+}
+
+func (r *rotatedFile) Write(bs []byte) (int, error) {
+	// Do the rotate check at most once a minStatInterval once any given file is
+	// open. Limits the number of stat() calls when there's a lot of logging,
+	// and limits the number of warnings if we're stuck unable to rotate for
+	// whatever reason.
+	if r.current != nil && time.Since(r.lastStat) >= r.minStatInterval {
+		info, err := os.Lstat(r.name)
+		if err != nil && !os.IsNotExist(err) {
```

Comment:
We are the ones writing to this file, why can't we just track the number of bytes we've written so far?

Reply:
We could do that, especially given we start from zero, so yeah, let me fix.